.github/workflows/build-custom-docker-image.yml

```diff
@@ -0,0 +1,38 @@
+name: OpenSearch Dashboards Build Custom Docker Image
```

Comment:
I'm wondering that should we create a template for all the docker image and this is just one case of docker image job.

Reply:
Yeah, I thought about that, since right now, we only have one use case, thus having it set like this, but later we can optimize it to be a general action/template and then we just have custom yaml to feed data to it